optuna/study/_optimize.py

```diff
@@ -189,6 +189,7 @@ def _run_trial(
         optuna.storages.fail_stale_trials(study)
 
     trial = study.ask()
+    study._thread_local.cache_all_trials.activate(lambda: study.get_trials(deepcopy=False))
```

Comment:
Just to clarify, is this functionality meant to replace `_CachedStorage`?
With `_CachedStorage` we only update non-cached (non-finished) trials. Here, we do that too if these two features co-exists, otherwise if it's to be replaced, we'll query the entire history in each cache update (every TTL interval).

Reply:
we don't meat to replace `_CachedStorage`. please have a look at this comment for our plan
https://github.com/optuna/optuna/pull/4366#discussion_r1100976416